iib/workers/tasks/build.py

```diff
@@ -720,6 +721,61 @@ def _verify_index_image(
         )
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str base_dir: the base directory where the database and index.Dockerfile are created
+    :param str bundle: the bundle path
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = "INSERT INTO properties (type, value, operatorbundle_name, operatorbundle_version, \
+        operatorbundle_path) VALUES (?, ?, ?, ?, ?);"
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property["type"],
+            property["value"],
+            property["operatorbundle_name"],
+            property["operatorbundle_version"],
+            property["operatorbundle_path"],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+# Check if the bundle is using deprecated APIs AND does not have the max OCP version set to 4.8
+def _requires_max_ocp_version(bundle):
+    """
+    Whether the bundle requires the olm.maxOpenShiftVersion property added to the db.
+
+    :param list bundles: a list of dicts representing bundles
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '-b',
+        'podman',
+    ]
+    result = run_cmd(cmd, strict=False)
+    if result:
+        output = json.loads(result)
+        if not output['passed']:
+            for msg in output['outputs']:
+                if "olm.maxOpenShiftVersion" in msg['message']:
```

Comment:
Maybe I'm misunderstanding this semantically. Is True returned when the bundle doesn't require this property to be added? Shouldn't it be the other way around?

Reply:
added some comments to clarify - hope it helps!